beacon-chain/rpc/core/validator.go

```diff
@@ -247,6 +254,47 @@ func (s *Service) SubmitSignedContributionAndProof(
 	return nil
 }
 
+func (s *Service) validateSignedContributionAndProof(ctx context.Context,
```

Comment:
much of the same steps as func (s *Service) rejectInvalidSyncAggregateSignature(m *ethpb.SignedContributionAndProof) validationFn maybe I can reuse...